tests/udp_socket.rs

```diff
@@ -848,3 +860,79 @@ pub fn multicast() {
         }
     }
 }
+
+#[test]
+fn udp_socket_edge_triggered_behavior() {
+    let (mut poll, mut events) = init_with_poll();
+
+    let socket1 = UdpSocket::bind(any_local_address()).unwrap();
+    let socket2 = UdpSocket::bind(any_local_address()).unwrap();
+    let socket3 = UdpSocket::bind(any_local_address()).unwrap();
+
+    let address1 = socket1.local_addr().unwrap();
+    let address3 = socket3.local_addr().unwrap();
+
+    poll.registry()
+        .register(&socket1, ID1, Interests::READABLE.add(Interests::WRITABLE))
+        .expect("unable to register UDP socket");
+    poll.registry()
+        .register(&socket2, ID2, Interests::READABLE.add(Interests::WRITABLE))
+        .expect("unable to register UDP socket");
+
+    expect_events(
+        &mut poll,
+        &mut events,
+        vec![
+            ExpectEvent::new(ID1, Interests::WRITABLE),
+            ExpectEvent::new(ID2, Interests::WRITABLE),
+        ],
+    );
+
+    {
+        socket2.connect(address3).unwrap();
+
+        expect_events(
```

Comment:
Connect doesn't generate event on most platforms. So I don't think this behaviour is required.

Reply:
This is also why the CI fails.

Reply:
reverted, not needed. Actually connect generates events on all platforms except Linux. CI failed on Linux only.

Reply:
Mio does not guarantee to generate events unless you get a WouldBlock :)

Reply:
Yep, love this Mio guarantee, makes things much clearer than before.